lib/connectors/base/connector.rb

```diff
@@ -33,11 +32,19 @@ def self.service_type
         raise 'Not implemented for this connector'
       end
 
-      def initialize(configuration: {})
+      attr_reader :rules, :advanced_filter_config
+
+      def initialize(configuration: {}, job_description: {})
         @configuration = configuration.dup || {}
+        @job_description = job_description.dup || {}
+
+        filter = get_filter(@job_description[:filtering])
+
+        @rules = Utility::Common.return_if_present(filter[:rules], [])
+        @advanced_filter_config = Utility::Common.return_if_present(filter[:advanced_config], {})
       end
 
-      def yield_documents; end
+      def yield_documents(job_description: {}); end
```

Comment:
```suggestion
      def yield_documents; end
```
Now that `@job_description` is available through the constructor, this is not necessary